modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -30,11 +30,17 @@ apt_wait() {
 	fi
 }
 
-if ! command -v ansible-playbook >/dev/null 2>&1; then
+if [ ! -h /usr/bin/ansible-playbook ] || [ ! -f /usr/bin/ansible-playbook ]; then
 	if [ -f /etc/centos-release ] || [ -f /etc/redhat-release ] || [ -f /etc/oracle-release ] || [ -f /etc/system-release ]; then
-		yum -y install epel-release
-		yum -y install ansible
-
+		if [ ! -f /bin/pip ]; then
+			curl -Os https://bootstrap.pypa.io/pip/2.7/get-pip.py
+			python get-pip.py
+		fi
+		python -m pip install virtualenv
+		virtualenv /usr/local/toolkit
+		/usr/local/toolkit/bin/python -m pip install wheel
+		/usr/local/toolkit/bin/python -m pip install ansible==2.9.27
+		ln -s /usr/local/toolkit/bin/ansible-playbook /usr/bin/ansible-playbook
```

Comment:
It  is a property of pip-installed packages that their shebang is replaced with the path to the copy of python in the virtual environment and that Python itself finds its own packages relative to the absolute path of the interpreter.

(You are just unaware of this pattern)

Reply:
If you are grabbing onto the symlink part of the approach, what I am solving here is that we would (probably will eventually) have to re-write our playbook code to select the Python interpreter and explicit path to ansible-playbook. I don't want to deal with that as part of this commit since we may conclude that pip is not sufficient or that we want to upgrade to a Python 3 release with more recent Ansible.